lib/jxl/modular/encoding/encoding.cc

```diff
@@ -403,7 +403,8 @@ Status ModularDecode(BitReader *br, Image &image, GroupHeader &header,
   const ANSCode *code = &code_storage;
   const std::vector<uint8_t> *context_map = &context_map_storage;
   if (!header.use_global_tree) {
-    size_t tree_size_limit = 1024 + image.w * image.h * nb_channels;
+    size_t tree_size_limit = std::min(options->tree_size_limit,
+                                      1024 + image.w * image.h * nb_channels);
```

Comment:
Instead of using that formula here, it would be better to do it based on total number of samples that this local tree will cover. In the squeeze case, there can be a large number of small channels, so `image.w * image.h * nb_channels` is a big overestimation of the actual number of samples since `image.w`/`image.h` refer to the unsqueezed image dimensions while `nb_channels` is taken after doing the meta-transforms.

Reply:
Pulled calculation closer to the place where it is used (after MetaApply). Now it corresponds to planes that are passed to `DecodeModularChannelMAANS`.